### DISCUSSION TITLE
how to find/delete playItem by file path?

### DISCUSSION THREAD

**Discussion by KnightMurloc** [DeaDBeeF-Player/deadbeef#2732] [Q&A] [answered]
I am writing a plugin and I need to remove a file from a playlist by path.

I tried something like this:

    ddb_playItem_t* current = deadbeef->plt_get_first(pl,0);

    deadbeef->pl_lock();
    while (current){

        const char* path = deadbeef->pl_find_meta(current,"location");
        printf("%p\n",path);
        ddb_playItem_t* tmp = current;
        current = deadbeef->pl_get_next(current,0);
        deadbeef->pl_item_unref(tmp);
    }
    deadbeef->pl_unlock();

but the key "location" and "path" always return null.
hence a few questions. how can i get the path from ddb_playItem_t? and what is the best way to find and remove such an element?

**Comment by Oleksiy-Yakovenko** [comment, marked as answer]
Most of the time, deadbeef uses ":URI" field for getting file paths.
Also worth noting, that for local paths the :URI field won't have the "file://" prefix
